scripts/cat/names.py

```diff
@@ -265,6 +265,25 @@ def __repr__(self):
         # then suffixes based on ages (fixes #2004, just trust me)
 
         # Handles suffix assignment with outside cats
+        if (
+            self.cat.status.is_lost(CatGroup.PLAYER_CLAN)
+            and not self.cat.status.is_former_clancat
+            and self.suffix
+        ):
+            # these are cats who were born to a parent who'd been lost frm their clan, and who's parent decided to keep with traditional naming
+            age_to_rank = {
+                CatAge.NEWBORN: CatRank.NEWBORN,
+                CatAge.KITTEN: CatRank.KITTEN,
+                CatAge.ADOLESCENT: CatRank.APPRENTICE,
+            }
+            rank = None
+            if self.cat.age in age_to_rank:
+                rank = age_to_rank[self.cat.age]
+            if rank:
+                return self.prefix + self.names_dict["special_suffixes"][rank]
```

Comment:
```suggestion
                rank = age_to_rank[self.cat.age]
                return self.prefix + self.names_dict["special_suffixes"][rank]
```
unless I'm misreading this, there's no way for `rank` to be non-null where the previous statement is `False`.